pkgs/desktops/gnome-3/extensions/default.nix

```diff
@@ -0,0 +1,25 @@
+{pkgs, ...}: let
+  # Index of all scraped extensions (with supported versions)
+  extensionsIndex = builtins.fromJSON (builtins.readFile ./extensions.json);
+  # Filter the extensions index to the matching Gnome shell version.
+  filterShellVersion = shell-version: extensions-index:
+    (builtins.map (e: {
+      inherit (e) uuid name description link pname;
```

Comment:
It appears that we are missing a license? How do we ensure we aren't violating someones license?

Reply:
That's a tricky question. The license is not annotated anywhere in the metadata. Some git repositories also apparently forgot to add one at all. I don't know if the extensions.gnome.org site does any licensing foo when uploading an extension.

Is there a "unknown" license attribute? We may want to play it safe and conservatively call all extensions "non-free" …

Reply:
Turns out it's all bright and shiny: https://wiki.gnome.org/Projects/GnomeShell/Extensions/Review#Licensing GPLv2+